File: Week-2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[]) {
    if (argc != 2) {
        printf("Usage: ./caesar key\n");
        return 1;
    }

    int key = atoi(argv[1]);
    
    if (key < 0) {
        printf("Key must be a non-negative integer.\n");
        return 1;
    }

    string message = get_string("Enter the message: ");

    int ASCII_A = 65;
    int ascii_a = 97;
    int len_alphabet = 26;

    printf("ciphertext: ");

    int n = strlen(message);

    for (int i = 0; i < n; i++) {
        if (isupper(message[i]))
            printf("%c", ((message[i] - ASCII_A + key) % len_alphabet + len_alphabet) % len_alphabet + ASCII_A);

        else if (islower(message[i]))
            printf("%c", ((message[i] - ascii_a + key) % len_alphabet + len_alphabet) % len_alphabet + ascii_a);

        else
            printf("%c", message[i]);
    }

    printf("\n");
    return 0;
}

File: Week-6/caesar.py
from sys import argv, exit
from cs50 import get_string

def main():
    if len(argv) != 2:
        print("Usage: python caesar.py key (where key is a non-negative integer)")
        exit(1)

    try:
        key = int(argv[1])
        if key < 0:
            raise ValueError("Key must be a non-negative integer.")
    except ValueError:
        print("Key must be a non-negative integer.")
        exit(1)

    plaintext = get_string("plaintext: ")

    print("ciphertext: ", end="")

    for char in plaintext:
        if not char.isalpha():
            print(char, end="")
            continue

        ascii_offset = 65 if char.isupper() else 97

        pi = ord(char) - ascii_offset
        ci = (pi + key) % 26

        print(chr(ci + ascii_offset), end="")

    print()

    return 0

if __name__ == "__main__":
    main()